### PR TITLE
Porting extending UI doc.

### PR DESCRIPTION
**Pull request by shinde-rahul**
Addressing https://github.com/mautic/developer-documentation-new/issues/72

<a href="https://gitpod.io/#https://github.com/mautic/developer-documentation-new/pull/80"></a>

### CODE REVIEW

File: docs/components/ui.rst [hunk 1]
```diff
@@ -1,3 +1,275 @@
...
+
+    <?php
+    $dropdownOpenHtml = '<button type="button" class="btn btn-default btn-nospin  dropdown-toggle" data-toggle="dropdown" aria-expanded="false"><i class="fa fa-caret-down"></i></button>'
+              ."\n";
+    $dropdownOpenHtml .= '<ul class="dropdown-menu dropdown-menu-right" role="menu">'."\n";
+
+    echo $view['buttons']->reset($app->getRequest(), 'custom_location')->renderButtons($dropdownOpenHtml, '</ul>');
+
+
+A Plugin can define it's own locations that other Plugins can leverage by using the template ``buttons`` helper.
```
Comment (RCheesley) [inline docs/components/ui.rst:259]:
```suggestion
A Plugin can define its own locations that other Plugins can leverage by using the template ``buttons`` helper.
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
Just one small grammar tweak! 🚀

**Review by dennisameling — APPROVED** [review]
This LGTM, thanks @shinde-rahul!

@RCheesley mind doing a final review? Note that Vale reports some suggestions, but it can't show those in the GitHub UI [due to a known issue](https://github.com/errata-ai/vale-action/issues/12) in the Vale action.

https://github.com/mautic/developer-documentation-new/actions/runs/3559535038/jobs/5978951313

**Comment by dennisameling** [comment]
Update: the Vale checks now run correctly on this PR: https://github.com/mautic/developer-documentation-new/actions/runs/3567127312/jobs/5994433686

@RCheesley want me to address those so we can get this PR merged? 😊

**Comment by RCheesley** [comment]
Yes please if you have the capacity!

**Review by dennisameling — APPROVED** [review]

**Comment by dennisameling** [comment]
Done @RCheesley. OK to merge?

**Review by RCheesley — APPROVED** [review]

**Pull request merged**